src/main/java/ru/spbstu/calculator/Constants.java

```diff
@@ -1,8 +1,16 @@
 package ru.spbstu.calculator;
 
+import java.util.HashSet;
+import java.util.Set;
+
 public class Constants {
     public static final int DEFAULT_WINDOW_SIZE = 10;
-    public static final Character[] NUCLEOTIDES = {'A', 'C', 'G', 'T'};
+    public static final Set<Character> NUCLEOTIDES = new HashSet<>(){{
+        add('A');
+        add('C');
+        add('T');
+        add('G');
+    }};
     public static final double frequency = 0.25f;
     public enum calculationMethod {
```

Comment:
Could you rename to CalculationMethod ? + I recommend you separating this enum from Constants